app/helpers/sharing_permissions_helper.rb

```diff
@@ -1,225 +1,51 @@
-    }
-
-    resource_items.each do |item|
-
-      entry_item = {
-          id: unique_node_id(item),
-          data: { loadable: false },
-          li_attr: { 'data-node-id' => node_id(item), class:"asset-node-row"},
-          a_attr: {class:"asset-node"},
-          children: [] ,
-          icon: asset_path(resource_avatar_path(item) || icon_filename_for_key("#{item.class.name.downcase}_avatar")),
-          text: "#{h(item.title)}  #{icon_link_to("", "new_window", item , options = {target:'blank',class:'asset-icon',:onclick => 'window.open(this.href, "_blank");'})}"
-      }
-
-      permissions_array = get_permission(item)
-      entry_item[:children] = permissions_array
-      parent[:children].append(entry_item)
-    end
-    parent
-  end
-
-
-  def create_policy_node(item, policy_text,sharing_policy_changed)
-
-    entry = {
-        id: unique_policy_node_id(item),
-        data: { loadable: false },
-        li_attr: { 'data-node-id' => "Permission-"+node_id(item), class:"hide_permission" },
-        a_attr: { class:"permission-node #{sharing_policy_changed}"},
-        children: [],
-        text:policy_text
-    }
-    entry
-  end
-
-  def get_permission (item)
-
-    policy =[]
+    policy = []
 
     # policy
     downloadable = item.try(:is_downloadable?)
-    policy_text = "#{Policy.get_access_type_wording(item.policy.access_type, downloadable)} by Public"
+    policy_text = "<strong>Public:</strong> #{Policy.get_access_type_wording(item.policy.access_type, downloadable)}."
     sharing_policy_changed = (@batch_sharing_permission_changed && (@items_for_sharing.include? item) && !@policy_params[:access_type].nil?)? "sharing_permission_changed" : ""
 
-    policy.append(create_policy_node(item,policy_text,sharing_policy_changed))
+    policy.append("<div id=\"#{item_id}-Permission-Policy\" class=\"permission-node #{sharing_policy_changed}\">#{policy_text}</div>".html_safe)
 
     #permission
-    option = {:onclick => 'window.open(this.href, "_blank");'}
-
+    option = { target: :_blank }
     item.policy.permissions.map do |permission|
+      sharing_policy_changed = @batch_sharing_permission_changed && (@items_for_sharing.include? item) ? PolicyHelper::permission_changed_item_class(permission, @policy_params) : ""
+      div_id = 'case-not-found'
       case permission.contributor_type
       when Permission::PROJECT
         m = Project.find(permission.contributor_id)
-        policy_text ="#{Policy.get_access_type_wording(permission.access_type, downloadable)} by Project  #{link_to(h(m.title), m, option)}"
-        sharing_policy_changed = @batch_sharing_permission_changed && (@items_for_sharing.include? item) ? PolicyHelper::permission_changed_item_class(permission, @policy_params) : ""
-        policy.append(create_policy_node(item,policy_text,sharing_policy_changed))
+        policy_text = "<strong>Project  #{link_to(h(m.title), m, option)}:</strong>"
+        div_id = "#{item_id}-Permission-Project-#{m.id}:<strong>"
```

Comment:
You could move this outside of the switch statement and use `permission.contributor_type` to generate the ID for all cases.

Also the `<strong>` at the end seems to be a mistake.

Reply:
Fixed in [ea2e182](https://github.com/seek4science/seek/pull/1551/commits/ea2e182157ee7b7c7306dae42fc14dafd349d85a)